### PR TITLE
Fixed major Sonar issues in different classes

### PR DESCRIPTION
**Pull request by vbergmann**

### CODE REVIEW

File: src/main/java/org/aludratest/testcase/data/impl/xml/TestCaseUtil.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+/** Provides utility methods for test case handling.
+ * @author Volker Bergmann */
+public class TestCaseUtil {
+
+    private TestCaseUtil() {
+    }
+
+    /** Creates a unique number for a test data set. */
+    public static String getNextAutoId(List<TestCaseData> dataSets, boolean error) {
```
Comment (vbergmann) [inline src/main/java/org/aludratest/testcase/data/impl/xml/TestCaseUtil.java]: It's here by accident, belongs to pull request #245

File: src/main/java/org/aludratest/testcase/data/TestCaseData.java [hunk 2]
```diff
@@ -134,7 +134,7 @@ public String getDescription() {
     public Data[] getData() {
         if (dataSource != null) {
             Data[] sourceData = dataSource.getData();
-            return (sourceData == null ? new Data[0] : ArrayUtil.copyOfRange(sourceData, 0, sourceData.length));
+            return ((sourceData == null) ? new Data[0] : ArrayUtil.copyOfRange(sourceData, 0, sourceData.length));
```
Comment (vbergmann) [inline src/main/java/org/aludratest/testcase/data/TestCaseData.java]: see above

File: src/main/java/org/aludratest/testcase/data/TestCaseData.java [hunk 3]
```diff
@@ -64,7 +64,7 @@ public TestCaseData(String id, String description, Data[] data) {
     public TestCaseData(String id, String description, Data[] data, boolean ignored) {
         this.id = id;
         this.description = description;
-        this.data = (data == null ? new Data[0] : ArrayUtil.copyOfRange(data, 0, data.length));
+        this.data = ((data == null) ? new Data[0] : ArrayUtil.copyOfRange(data, 0, data.length));
```
Comment (vbergmann) [inline src/main/java/org/aludratest/testcase/data/TestCaseData.java]: Sonar wants us to use brackets to indicate operator priority explicitly

File: src/main/java/org/aludratest/testcase/data/impl/xml/TestCaseUtil.java (hunk 1 above)
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/testcase/data/impl/xml/TestCaseUtil.java]: Could you then please remove it from this PR? Thanks!

File: src/main/java/org/aludratest/testcase/data/TestCaseData.java (hunk 3 above)
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/testcase/data/TestCaseData.java]: But brackets around the **whole** expression should be treated superflouous even by Sonar: compare `x = ((a && b) ? c : d)` vs `x = (a && b) ? c : d`

**Pull request merged**